packages/sdk/client-services/src/packlets/invitations/invitations-service.ts

```diff
@@ -72,6 +90,22 @@ export class InvitationsServiceImpl implements InvitationsService {
     });
   }
 
+  async getPersistentInvitations(): Promise<GetPersistentInvitationsResponse> {
+    const persistentInvitations = this._metadataStore.getPersistentInvitations();
+
+    // get saved persistent invitations, filter and remove from storage those that have expired.
+    const freshInvitations = persistentInvitations.filter(async (invitation) => {
```

Comment:
Meta: Given the scale, this is of course, but in general I might have a separate function to flush expired invitations (something that can be polled periodically).

And in general possibly `removeInvitations(ids)` so that you first get a filtered set (synchronously) then atomically delete them all in one call.